tests/test_simpleion.py

```diff
@@ -318,6 +318,23 @@ def equals():
         assert ion_equals(p.obj, res)  # Redundant, but provides better error message.
 
 
+@parametrize(
+    *tuple(chain(
+        generate_scalars_text(SIMPLE_SCALARS_MAP_TEXT),
+        generate_containers_text(_SIMPLE_CONTAINER_MAP),
+        generate_annotated_values_text(SIMPLE_SCALARS_MAP_TEXT, _SIMPLE_CONTAINER_MAP),
+    ))
+)
+def test_dumps(p):
+    # test dumps
+    res = dumps(p.obj, sequence_as_stream=p.stream)
+    if not p.has_symbols:
+        assert (b'$ion_1_0 ' + p.expected) == res
```

Comment:
Per the [Travic CI build](https://travis-ci.org/amzn/ion-python/builds/285880665?utm_source=github_status&utm_medium=notification), this fails on Python 3, probably because `res` is a text (unicode) type while the left hand side is a binary type (bytes). You may need to decode the left hand side.

The [README](https://github.com/amzn/ion-python/blob/master/README.md) contains instructions for setting up `tox` to run local builds for multiple python interpreter versions.

Reply:
You're right. Decoding the left side resolves it.

As the README says, I'm having some issues getting pyenv to do things on macOS, but I was able to run tests manually on 2.7.10 and 3.6.3 installations and they're all green. I'll let Travis CI do its thing.